main.go

```diff
@@ -1,94 +0,0 @@
-// time-stamp microservice receives timestamps and returns json form of natural/unix timestamp.
-// Exercise by Free Code Camp.
-package main
-
-import (
-	"encoding/json"
-	"fmt"
-	"net/http"
-	"os"
-	"strconv"
-	"strings"
-	"time"
-)
-
-// Date struct returned as json.
-type Date struct {
-	Unix    *int64  `json:"unix"`
-	Natural *string `json:"natural"`
-}
-
-func main() {
-	http.HandleFunc("/", timestamp)
-
-	server := http.Server{
-		Addr: ":" + os.Getenv("PORT"),
-	}
-	fmt.Println("Listening on [", server.Addr, "]...")
-	err := server.ListenAndServe()
-	if err != nil {
-		panic(err)
-	}
-}
-
-func timestamp(w http.ResponseWriter, r *http.Request) {
-
-	tstr := r.URL.Path[1:]
-
-	w.Header().Set("Content-Type", "application/json")
-	w.Header().Set("charset", "utf-8")
-
-	t, err := returnTime(tstr)
-	if err != nil {
-		nilJSON, err := json.Marshal(Date{nil, nil})
-		if err != nil {
-			http.Error(w, "Json string failed to write", 500)
```

Comment:
Here, you're shadowing the error twice, first from `returnTime`, and second from `json.Marshal`. 
If either of the errors fail, it would be extremely hard to know why. 

You could instead return `http.Error(w, fmt.Sprintf("Json string failed to write, err: %q" err), http.StatusInternalServerError`)

My second nit here is that you should be using `http.Status...` instead of numeric codes.